File: scratch/api_test.py
import urllib.request
import json

base_url = 'http://localhost:8000/api'

def post(url, data, token=None):
    req = urllib.request.Request(url, data=json.dumps(data).encode('utf-8'))
    req.add_header('Content-Type', 'application/json')
    req.add_header('Accept', 'application/json')
    if token:
        req.add_header('Authorization', f'Bearer {token}')
    try:
        with urllib.request.urlopen(req) as response:
            return response.getcode(), json.loads(response.read().decode('utf-8'))
    except urllib.error.HTTPError as e:
        return e.code, json.loads(e.read().decode('utf-8'))

def get(url, token=None):
    req = urllib.request.Request(url)
    req.add_header('Accept', 'application/json')
    if token:
        req.add_header('Authorization', f'Bearer {token}')
    try:
        with urllib.request.urlopen(req) as response:
            return response.getcode(), json.loads(response.read().decode('utf-8'))
    except urllib.error.HTTPError as e:
        return e.code, json.loads(e.read().decode('utf-8'))

print("1. Logging in...")
code, body = post(f"{base_url}/login", {'email': 'tes@yaya.com', 'password': 'tes1234567'})
if code != 200:
    print(f"Login failed: {code} {body}")
    exit(1)

token = body.get('token') or body.get('access_token')
if not token and 'data' in body:
    token = body['data'].get('token') or body['data'].get('access_token')

print("Login OK")

journal_id = 9
journal_slug = 'test-journal'
print(f"2. Using Journal ID: {journal_id}")

print("3. Creating Submission...")
sub_data = {
    'journal_id': journal_id,
    'title': 'Runtime Verification Submission — Author Workspace',
    'abstract': 'This is a runtime verification submission created through the existing HexaLMS Author Workspace workflow.',
    'keywords': ['runtime verification', 'author workspace', 'hexalms'],
    'authors': [
        {
            'first_name': 'Tes',
            'last_name': 'User',
            'email': 'tes@yaya.com',
            'affiliation': 'Hexa',
            'is_corresponding': True
        }
    ]
}
code, body = post(f"{base_url}/submissions", sub_data, token)
if code != 201:
    print(f"Failed to create submission: {code} {body}")
    exit(1)

sub_id = body.get('data', {}).get('id') or body.get('id')
print(f"Created Submission ID: {sub_id}")

print("4. Verifying /author/submissions via API...")
code, body = get(f"{base_url}/submissions", token)
if code != 200:
    print(f"Failed to fetch submissions: {code} {body}")
    exit(1)

found = any(s['id'] == sub_id for s in body.get('data', []))
print(f"Submission in list: {'YES' if found else 'NO'}")

print("5. Verifying Submission Detail...")
code, body = get(f"{base_url}/submissions/{sub_id}", token)
print(f"Detail Code: {code}")
print(f"Detail Title: {body['data']['title']}")
print(f"Detail Status: {body['data']['status']}")

print("6. Check Editorial Visibility (Draft shouldn't be fully visible if restricted)")
code, body = get(f"{base_url}/user/journals/{journal_slug}/management/submissions", token)
print(f"Editorial access (should be 403, 404, or empty array since it's an author): {code}")

File: scratch/api_test.php
<?php
$baseUrl = 'http://localhost:8000/api';

function post($url, $data, $token = null) {
    $ch = curl_init($url);
    $headers = ['Content-Type: application/json', 'Accept: application/json'];
    if ($token) $headers[] = "Authorization: Bearer $token";
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_POST, true);
    curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($data));
    $resp = curl_exec($ch);
    $code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);
    return ['code' => $code, 'body' => json_decode($resp, true)];
}

function get($url, $token = null) {
    $ch = curl_init($url);
    $headers = ['Accept: application/json'];
    if ($token) $headers[] = "Authorization: Bearer $token";
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    $resp = curl_exec($ch);
    $code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);
    return ['code' => $code, 'body' => json_decode($resp, true)];
}

echo "1. Logging in...\n";
$login = post("$baseUrl/login", ['email' => 'tes@yaya.com', 'password' => 'tes1234567']);
if ($login['code'] !== 200) {
    die("Login failed: " . json_encode($login));
}
$token = $login['body']['token'] ?? $login['body']['data']['token'] ?? null;
if (!$token) {
    // try to find token in output
    if(isset($login['body']['access_token'])) {
        $token = $login['body']['access_token'];
    } elseif (isset($login['body']['data']['access_token'])) {
        $token = $login['body']['data']['access_token'];
    } else {
        die("Token not found in response: " . json_encode($login));
    }
}
echo "Login OK\n";

echo "2. Fetching Journals...\n";
$journals = get("$baseUrl/journals", $token);
if ($journals['code'] !== 200) {
    die("Failed to fetch journals: " . json_encode($journals));
}
$journalId = $journals['body']['data'][0]['id'] ?? null;
if (!$journalId) die("No journals found");
echo "Selected Journal ID: $journalId\n";

echo "3. Creating Submission...\n";
$subData = [
    'journal_id' => $journalId,
    'title' => 'Runtime Verification Submission — Author Workspace',
    'abstract' => 'This is a runtime verification submission created through the existing HexaLMS Author Workspace workflow.',
    'keywords' => ['runtime verification', 'author workspace', 'hexalms'],
    'authors' => [
        [
            'first_name' => 'Tes',
            'last_name' => 'User',
            'email' => 'tes@yaya.com',
            'affiliation' => 'Hexa',
            'is_corresponding' => true
        ]
    ]
];
$create = post("$baseUrl/submissions", $subData, $token);
if ($create['code'] !== 201) {
    die("Failed to create submission: " . json_encode($create));
}
$subId = $create['body']['data']['id'] ?? $create['body']['id'] ?? null;
echo "Created Submission ID: $subId\n";

echo "4. Verifying /author/submissions via API...\n";
$list = get("$baseUrl/submissions", $token);
if ($list['code'] !== 200) {
    die("Failed to fetch submissions: " . json_encode($list));
}
$found = false;
foreach ($list['body']['data'] as $s) {
    if ($s['id'] === $subId) {
        $found = true;
        break;
    }
}
echo "Submission in list: " . ($found ? "YES" : "NO") . "\n";

echo "5. Verifying Submission Detail...\n";
$detail = get("$baseUrl/submissions/$subId", $token);
echo "Detail Code: " . $detail['code'] . "\n";
echo "Detail Title: " . $detail['body']['data']['title'] . "\n";
echo "Detail Status: " . $detail['body']['data']['status'] . "\n";

